examples/js/loaders/OBJLoader.js

```diff
@@ -260,72 +267,103 @@ THREE.OBJLoader = ( function () {
 
 			},
 
-			addFace: function ( a, b, c, d, ua, ub, uc, ud, na, nb, nc, nd ) {
+			addFace: function ( args ) {
 
 				var vLen = this.vertices.length;
+				var uvLen = this.uvs.length;
+				var nLen = this.normals.length;
 
-				var ia = this.parseVertexIndex( a, vLen );
-				var ib = this.parseVertexIndex( b, vLen );
-				var ic = this.parseVertexIndex( c, vLen );
-				var id;
+				var ivs = [ ], ius = [ ], ins = [ ];
 
-				if ( d === undefined ) {
+				var mode = args.pop();
 
-					this.addVertex( ia, ib, ic );
+				switch ( mode ) {
 
-				} else {
+					case MODE.ADDING_VERTEX:
 
-					id = this.parseVertexIndex( d, vLen );
+						for ( var i = 0; i < args.length; i ++ ) {
 
-					this.addVertex( ia, ib, id );
-					this.addVertex( ib, ic, id );
+							if ( ! args[ i ] ) {
 
-				}
+								break;
 
-				if ( ua !== undefined ) {
+							}
 
-					var uvLen = this.uvs.length;
+							ivs.push( this.parseVertexIndex( args[ i ], vLen ) );
 
-					ia = this.parseUVIndex( ua, uvLen );
-					ib = this.parseUVIndex( ub, uvLen );
-					ic = this.parseUVIndex( uc, uvLen );
+						}
 
-					if ( d === undefined ) {
+						break;
 
-						this.addUV( ia, ib, ic );
+					case MODE.ADDING_VERTEX_UV:
 
-					} else {
+						for ( var i = 0; i < args.length - 1; i = i + 2 ) {
 
-						id = this.parseUVIndex( ud, uvLen );
+							if ( ! args[ i ] ) {
 
-						this.addUV( ia, ib, id );
-						this.addUV( ib, ic, id );
+								break;
 
-					}
+							}
 
-				}
+							ivs.push( this.parseVertexIndex( args[ i ], vLen ) );
+							ius.push( this.parseUVIndex( args[ i + 1 ], uvLen ) );
 
-				if ( na !== undefined ) {
+						}
 
-					// Normals are many times the same. If so, skip function call and parseInt.
-					var nLen = this.normals.length;
-					ia = this.parseNormalIndex( na, nLen );
+						break;
 
-					ib = na === nb ? ia : this.parseNormalIndex( nb, nLen );
-					ic = na === nc ? ia : this.parseNormalIndex( nc, nLen );
+					case MODE.ADDING_VERTEX_UV_NORMAL:
 
-					if ( d === undefined ) {
+						for ( var i = 0; i < args.length - 2; i = i + 3 ) {
 
-						this.addNormal( ia, ib, ic );
+							if ( ! args[ i ] ) {
 
-					} else {
+								break;
 
-						id = this.parseNormalIndex( nd, nLen );
+							}
 
-						this.addNormal( ia, ib, id );
-						this.addNormal( ib, ic, id );
+							ivs.push( this.parseVertexIndex( args[ i ], vLen ) );
+							ius.push( this.parseUVIndex( args[ i + 1 ], uvLen ) );
+							ins.push( this.parseNormalIndex( args[ i + 2 ], nLen ) );
 
-					}
+						}
+
+						break;
+
+					case MODE.ADDING_VERTEX_NORMAL:
+
+						for ( var i = 0; i < args.length - 1; i = i + 2 ) {
+
+							if ( ! args[ i ] ) {
+
+								break;
+
+							}
+
+							ivs.push( this.parseVertexIndex( args[ i ], vLen ) );
+							ins.push( this.parseNormalIndex( args[ i + 1 ], nLen ) );
+
+						}
+
+						break;
+
+				}
+
+				for ( var j = 2; j < ivs.length; j++ ) {
```

Comment:
```
j ++
```
Please use http://zz85.github.io/mrdoobapproves/ to check your formatting so someone else doesn't have to do it.